src/Compilers/Core/Portable/CommandLine/AnalyzerConfigSet.cs

```diff
@@ -562,52 +562,63 @@ private Section GetSection(string sectionName)
                 foreach ((var key, var value) in section.Properties)
                 {
-                    if (isGlobalSection && Section.PropertiesKeyComparer.Equals(key, GlobalKey))
+                    if (isGlobalSection && (Section.PropertiesKeyComparer.Equals(key, GlobalKey) || Section.PropertiesKeyComparer.Equals(key, GlobalLevelKey)))
                     {
                         continue;
                     }
 
-                    bool keyInSection = sectionDict.ContainsKey(key);
-                    bool keyDuplicated = duplicateDict?.ContainsKey(key) ?? false;
+                    bool keyInSection = sectionDict.TryGetValue(key, out var sectionValue) == true;
 
-                    // if this key is neither already present, or already duplicate, we can add it
-                    if (!keyInSection && !keyDuplicated)
+                    (int globalLevel, ArrayBuilder<string> configPaths) duplicateValue = default;
+                    bool keyDuplicated = duplicateDict?.TryGetValue(key, out duplicateValue) == true;
+
+                    if ((!keyInSection && !keyDuplicated)   // this key is neither present nor a duplicate, we can add it
+                        || (keyInSection && sectionValue.globalLevel < globalLevel) // this key should override the existing section one
+                        || (keyDuplicated && duplicateValue.globalLevel < globalLevel)) // this key should override previous duplicates
                     {
-                        sectionDict.Add(key, (value, configPath));
+                        sectionDict[key] = (value, configPath, globalLevel);
+
+                        if (keyDuplicated)
+                        {
+                            duplicateDict!.Remove(key);
+                        }
                     }
-                    else
+                    else if ((keyDuplicated && duplicateValue.globalLevel == globalLevel) // this key is another duplicate at the same level
+                            || (keyInSection && sectionValue.globalLevel == globalLevel)) // this key is a duplicate of a current section key
                     {
                         if (duplicateDict is null)
                         {
-                            duplicateDict = ImmutableDictionary.CreateBuilder<string, ArrayBuilder<string>>(Section.PropertiesKeyComparer);
+                            duplicateDict = ImmutableDictionary.CreateBuilder<string, (int, ArrayBuilder<string>)>(Section.PropertiesKeyComparer);
                             _duplicates.Add(section.Name, duplicateDict);
                         }
 
                         // record that this key is now a duplicate
-                        ArrayBuilder<string> configList = keyDuplicated ? duplicateDict[key] : ArrayBuilder<string>.GetInstance();
+                        ArrayBuilder<string> configList = duplicateValue.configPaths ?? ArrayBuilder<string>.GetInstance();
```

Comment:
>. [](start = 72, length = 1)

`?.`

Reply:
`duplicateValue` is a tuple, so it's never `null`, just the `configPaths` it contains when we didn't find it in the dictionary.